modules/aws-http4s/src/smithy4s/aws/json/AwsJsonCodecAPI.scala

```diff
@@ -16,8 +16,27 @@
 
 package smithy4s.aws.json
 
-private[aws] class AwsJsonCodecAPI()
-    extends smithy4s.http.json.JsonCodecAPI(
+import smithy4s.HintMask
+import smithy4s.http.json.JCodec
+import smithy4s.http.HttpMediaType
+
+private[aws] class AwsJsonCodecAPI(
+    httpMediaType: HttpMediaType.Type = HttpMediaType("application/json"),
+    hintMask: Option[HintMask] = None
+) extends smithy4s.http.json.JsonCodecAPI(
       new AwsSchemaVisitorJCodec(_),
-      None
-    ) {}
+      hintMask
+    ) {
+
+  override def writeToArray[A](codec: Codec[A], value: A): Array[Byte] = {
+    // AWS expects an empty object to be sent by clients even when the data
```

Comment:
is this related to the changes I've made in: https://github.com/disneystreaming/smithy4s/pull/1022/files#diff-3d68306dcc7da993cfe2a9162443c1ff9993076618cc20aaeaa58e8244753c8cL90

after the changes in the alloy spec: https://github.com/disneystreaming/alloy/pull/96/files